### PR TITLE
add Ftp file loading

### PR DESCRIPTION
**Pull request by hainm**

### CODE REVIEW

**Comment by swails** [comment]
Ah cool.  If `urlopen` works for files on an FTP server, then this is a nice solution (the PDB/CIF downloader uses `ftplib`).

There are a couple other places where URLs are handled separately from local file names -- a quick `grep` shows `parmed/amber/netcdffiles.py` and `parmed/amber/amberformat.py`.  If you add the `ftp://` condition to the URL handlers there, I'd be happy to merge this :).

Thanks!

**Comment by hainm** [comment]
Sure

Hai

**Comment by hainm** [comment]
@swails 

add "ftplib" too?

**Comment by swails** [comment]
> add "ftplib" too?

No -- `ftplib` is a Python module in the standard library that allows for interacting with FTP servers.

File: parmed/amber/netcdffiles.py [hunk 1]
```diff
@@ -200,6 +200,7 @@ def id_format(filename):
         Remote NetCDF files cannot be loaded
         """
         if filename.startswith('http://') or filename.startswith('https://'):
+            or filename.startswith('ftp://') or filename.startswith('ftplib'):
```
Comment (swails) [inline parmed/amber/netcdffiles.py]: `ftplib` is a Python module, not a filename prefix.  Also, this is a `SyntaxError`.

File: parmed/amber/netcdffiles.py [hunk 2]
```diff
@@ -543,7 +544,8 @@ def id_format(filename):
         -----
         Remote NetCDF files cannot be loaded
         """
-        if filename.startswith('http://') or filename.startswith('https://'):
+        if filename.startswith('http://') or filename.startswith('https://')
+            or filename.startswith('ftp://') or filename.startswith('ftplib'):
             return False
```
Comment (swails) [inline parmed/amber/netcdffiles.py]: Same as above (and still a syntax error)

**Comment by hainm** [comment]
don't have 'ftp' test cases for amber and netcdf file yet (since i can not find one). :D

**Comment by swails** [comment]
> don't have 'ftp' test cases for amber and netcdf file yet (since i can not find one). :D

That's fine -- it all uses `urlopen`, anyway.

File: parmed/amber/netcdffiles.py (hunk 2 above)
Comment (hainm) [inline parmed/amber/netcdffiles.py]: yeah, already has new commits :P

**Comment by swails** [comment]
Once Travis passes, I'll merge this.

**Pull request merged**